src/interchange/src/text_binary.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use bytes::BytesMut;
+use mz_repr::{ColumnName, ColumnType, RelationDesc};
+
+use crate::encode::{column_names_and_types, Encode};
+use crate::envelopes;
+
+#[derive(Debug)]
+pub enum TextBinaryEncoding {
```

Comment:
This feels like two unrelated encoders merged into one. Do they have any common parts and need to be in the same struct? The encoder below has the same structure as the `Json` encoder, is there maybe a `SimpleEncoding` enum that can host text/binary/json? Not sure. My inclination would be to have separate encoders per format.